### DISCUSSION TITLE
Guidance needed.

### DISCUSSION THREAD

**Discussion by marthiniusj** [AutoMapper/AutoMapper.Extensions.OData#153] [Q&A]
Good day.

I’m hoping someone could help me out.

I’ve forked this repo and modified it a bit to illustrate the problem I’m having, please feel free to pull and run it yourself.
https://github.com/marthiniusj/AutoMapper.Extensions.OData

We’re trying to make use of the ProjectionSetting Parameter property to apply additional filters on our mapping profiles.

Have a look at the OptSentantController class, line 62 to 73.
https://github.com/marthiniusj/AutoMapper.Extensions.OData/blob/master/WebAPI.OData.EFCore/Controllers/OpsTenantController.cs

![1](https://user-images.githubusercontent.com/79978614/191743797-e5977aaf-077f-4893-935a-38cf179f16f5.png)
 
Then have a look at the CoreBuildingMappings class, line 21.
https://github.com/marthiniusj/AutoMapper.Extensions.OData/blob/master/WebAPI.OData.EFCore/Mappings/CoreBuildingMappings.cs

![2](https://user-images.githubusercontent.com/79978614/191743824-6921cc09-c4bb-41bf-9aac-a60f8b0c16e3.png)

I then execute the following URL and I’m getting back the results shown below.

http://localhost:16324/opscity?$expand=Builders($expand=Buildings;$filter=Buildings/any())&$filter=Builders/any()
 
![3](https://user-images.githubusercontent.com/79978614/191743863-c427d1f0-272b-4e44-9cfb-5f55fc8f7460.png)

The mapping filter kicks in, because I’m only seeing the builder that I want, which is builderId = 1 (John in green) for the first city, London.

But the root /any() filter breaks (yellow block), because I still get back the second city, Leeds, even though the builders are empty.

So somehow, my mapping filter and the builders/any() filter don’t work well together.

Could you please show me what I’m doing wrong and how else I could achieve my goal of adding extra filters on the profile/controller level?

**Comment by BlaiseD** [comment]
There are tests e.g. [here](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/8b5d4896adc9404ea7ec581826b3327665062b15/AutoMapper.OData.EFCore.Tests/GetQueryTests.cs#L853) showing the expansions filter working independently of the root filter.  I see your scenario includes a filter in the map using plus a parameter.

Maybe you can isolate the one thing causing the unexpected result?
